webhooks/jira: Update docs to conform to style guide.

eeshangarg opened the pull request:
I spend a lot of time on this. One of our users had reported that
this webhook wasn't working at all. So I tested this with a local
ngrok instance and made sure that it was working. I also took this
opportunity to rewrite the docs for this, which were quite outdated.

@rishig: FYI another one down! :)

eeshangarg commented:
@timabbott: Mentioning you since we recently had a conversation about this one. :)

timabbott commented on zerver/webhooks/jira/doc.md:
```diff
@@ -1,20 +1,33 @@
+1. {!create-bot-construct-url-indented.md!}
 
-[1]: https://developer.atlassian.com/jiradev/jira-apis/webhooks#Webhooks-jiraadmin
+1. Go to your **Site administration** page, and click on **Jira** on the left.
```
This should have them start with logging into Jira, right?

eeshangarg replied:
Rishi recommended that we should leave out superfluous and obvious steps like "Log in to X" and assume that the user is logged in.

rishig replied:
We do want them to know that we're switching from Zulip to Jira though. We can do:
"Go to your Jira **Site administration** page. Click on **Jira** on the left. .."

rishig commented:
Thanks @eeshangarg! This was an important one.

We should use the current product names for the various JIRA cloud things mentioned. E.g. "JIRA OnDemand" I think no longer exists, and we should use Atlassian's name for whatever their hosted JIRA is called. 

Made a few other edits; copying the updated text below:
---------

Get Zulip notifications for your JIRA projects!

These instructions apply to Atlassian's hosted JIRA, and JIRA Server version
5.2 or greater. For older installs, you'll need our [JIRA Plugin](./jira-plugin).

1. {!create-stream.md!}

1. {!create-bot-construct-url-indented.md!}

1. Go to your JIRA **Site administration** page. Click **Jira** on the left.
   On the left sidebar, scroll down, and under **Advanced**, click **WebHooks**.
   Click **+ Create a WebHook**.

1. Set **Name** to a name of your choice, such as `Zulip`. Set **Status** to
   **Enabled**, and set **URL** to the URL constructed above. Select the events
   you'd like to be notified about, and click **Create**. We currently
   support the following events:
    * when an issue is created
    * when an issue is deleted
    * when an issue is updated

!!! tip ""
    If you are using the OnDemand hosted JIRA, follow the instructions
    [on the Atlassian wiki][1] for locating the Webhook UI.

[1]: https://developer.atlassian.com/jiradev/jira-apis/webhooks#Webhooks-jiraadmin

{!congrats.md!}

![](/static/images/integrations/jira/001.png)

eeshangarg commented:
@timabbott: Thanks for the review!

@rishig: Thanks for the review! Incorporated your changes and I also changed the reference to "OnDemand" to "Atlassian Cloud" (which is what it is called now)! 

Happy to help! Have a great day! :)

rishig commented:
Is there a difference between "Atlassian's hosted JIRA" in the intro and "Atlassian Cloud"?

eeshangarg commented:
@rishig: Well, there isn't really! Also, I did some more digging and the tip at the bottom (the way it is phrased) is highly redundant, so I just changed the tip to refer to Atlassian's docs for people who are more curious about how JIRA webhooks work!

If you ask Tim, you'll find that I rant about Atlassian quite a lot. The main issue I have with them is that in their efforts (efforts that could as easily be characterized as "haste") to acquire other projects/companies, they rename their projects and jumble the fine print so often that every product has 5 different versions of the same documentation, so you never know where to go! This is especially true about BitBucket and JIRA, which have like 3 different versions with a dozen different names! And so on...

But it should be all good now! Thanks for your guidance! :)

\-Eeshan

rishig commented:
It's not clear (after a few minutes of reading) what the connection is between the atlassian documentation and our documentation. Is it so the user can learn about scope?

rishig commented:
Merged, thanks @eeshangarg!
